onnxruntime/contrib_ops/cpu/gather_nd.h

```diff
@@ -42,8 +42,8 @@ class GatherND final : public OpKernel, protected GatherNDBase {
   explicit GatherND(const OpKernelInfo& info) : OpKernel(info) {}
   Status Compute(OpKernelContext* context) const override;
 private:
-  Status GatherNumber(const Prepare& p) const;
-  Status GatherString(const Prepare& p) const;
+  Status GatherNumber(const Prepare& p, onnxruntime::concurrency::ThreadPool* ttp) const;
+  Status GatherString(const Prepare& p, onnxruntime::concurrency::ThreadPool* ttp) const;
```

Comment:
Given that the ThreadPool will always be created, using a ptr for it is probably misleading giving the impression that it could be nullptr. Passing by reference is better. You can choose to clean it up in a different PR.